src/compiler/declarationEmitter.ts

```diff
@@ -327,7 +327,7 @@ namespace ts {
             }
             else {
                 errorNameNode = declaration.name;
-                resolver.writeTypeOfDeclaration(declaration, enclosingDeclaration, TypeFormatFlags.UseTypeOfFunction, writer);

```

Comment:
you need to do this in all places where we call `resolver.write*`, e.g. `writeReturnTypeAtSignature`